how to pass `-std=c++17` to company-clang for version specific completion?

consider this minimal example
````
1 #include<iostream>
2 #include<vector>
3 int main () {
4   std::vector<int> g{3,4,5};
5   std::cout<<std::size(g)<<std::endl;
6   return 0;
7 }
````
from the commandline **and** within `company-mode`  I get the same error
````
:~$ /usr/bin/clang -fsyntax-only -Xclang -code-completion-macros -x c++ a.cpp
a.cpp:5:19: error: no member named 'size' in namespace 'std'
  std::cout<<std::size(g)<<std::endl;
             ~~~~~^
1 error generated.
````
alsoin the `clang-error`  buffer but i was able to do other completions `g.emplace_back()` and `vector` methods.

 back in the commandline
````
~$ /usr/bin/clang -std=c++17 -fsyntax-only -Xclang -code-completion-macros -x c++ a.cpp
````
works and compiles correctly.

I was under the impression making ` (setq company-clang-arguments "-std=c++17")` would fix this but  I got
````
Wrong type argument: stringp, 45
 Company: An error occurred in auto-begin
````
when i tried an `std::si` and tried to complete it for `Company version: 0.9.10`.   I looked at

````
(defsubst company-clang--build-complete-args (pos)
  (append '("-fsyntax-only" "-Xclang" "-code-completion-macros")
          (unless (company-clang--auto-save-p)
            (list "-x" (company-clang--lang-option)))
          company-clang-arguments
          (when (stringp company-clang--prefix)
            (list "-include" (expand-file-name company-clang--prefix)))
          (list "-Xclang" (format "-code-completion-at=%s"
                                  (company-clang--build-location pos)))
          (list (if (company-clang--auto-save-p) buffer-file-name "-"))))
````
and can see `company-clang-arguments` being used to build the parameters passed to `clang` but it errors out for me if I add `-std=c++17`.

so, `std::size()` which is part of `include<iterator` is not completing for me.

--- a/company-clang.el
+++ b/company-clang.el
@@ -39,9 +39,10 @@
   :type 'file)
 
 (defcustom company-clang-begin-after-member-access t
-  "When non-nil, automatic completion will start whenever the current
-symbol is preceded by \".\", \"->\" or \"::\", ignoring
-`company-minimum-prefix-length'.
+  "When non-nil, start automatic completion after member access operators.
+
+Automatic completion starts whenever the current symbol is preceded by
+\".\", \"->\" or \"::\", ignoring `company-minimum-prefix-length'.
 
 If `company-begin-commands' is a list, it should include `c-electric-lt-gt'
 and `c-electric-colon', for automatic completion right after \">\" and
@@ -59,7 +60,7 @@ it.  That allows the flags use relative file names within the project."
   :safe 'booleanp)
 
 (defcustom company-clang-arguments nil
-  "Additional arguments to pass to clang when completing.
+  "A list of additional arguments to pass to clang when completing.
 Prefix files (-include ...) can be selected with `company-clang-set-prefix'
 or automatically through a custom `company-clang-prefix-guesser'."
   :type '(repeat (string :tag "Argument")))

